sql/core/src/main/scala/org/apache/spark/sql/api/r/SQLUtils.scala

```diff
@@ -247,7 +247,7 @@ private[sql] object SQLUtils extends Logging {
     dataType match {
       case 's' =>
         // Read StructType for DataFrame
-        val fields = SerDe.readList(dis).asInstanceOf[Array[Object]]
+        val fields = SerDe.readList(dis, jvmObjectTracker = null).asInstanceOf[Array[Object]]
```

Comment:
The two changes below are functions used in serializing and deserializing StructType from R to JVM (when calling collect or parallelize)

Again I think since StructTypes themselves can't contain other JVM references I think its fine, but I'm less sure about this. (i.e what happens when we say embed an array into the structType). 

Another option here is to figure out which RBackend is calling this function and use its JVMObjectTracker ? Do we want to support cases where there are multiple active RBackends ? If not we can denote an `activeRBackend` and then use its jvm object tracker in these places ?